### PR TITLE
Correct `Camera.mode` command.

### PR DESCRIPTION
**Pull request by NissimHadar**
# Description
The camera command mode (e.g. Camera.mode = "first person") was not functioning correctly.  This was a timing issue that was corrected by calling the relevant code only when needed.

I have labeled as "Scripting API change" as this PR may be changing behaviour that is expected.

# Testing
Test is: 
https://highfidelity.testrail.net/index.php?/cases/view/17447

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10041/HighFidelity-Beta-PR13220-06203d44d8b52b9aad0b9e984eaee05d1a660c9a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10041/HighFidelity-Beta-PR13220-06203d44d8b52b9aad0b9e984eaee05d1a660c9a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10041/HighFidelity-Beta-Interface-PR13220-06203d44d8b52b9aad0b9e984eaee05d1a660c9a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10041/HighFidelity-Beta-PR13220-06203d44d8b52b9aad0b9e984eaee05d1a660c9a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10041/HighFidelity-Beta-Interface-PR13220-06203d44d8b52b9aad0b9e984eaee05d1a660c9a.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10042/HighFidelity-Beta-PR13220-e07e716d4051930daab3da99d40ef8aa42903efa.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10042/HighFidelity-Beta-Interface-PR13220-e07e716d4051930daab3da99d40ef8aa42903efa.dmg

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10042/HighFidelity-Beta-Interface-PR13220-e07e716d4051930daab3da99d40ef8aa42903efa.exe

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -2248,6 +2248,11 @@ void MyAvatar::updateActionMotor(float deltaTime) {
     float boomChange = getDriveKey(ZOOM);
     _boomLength += 2.0f * _boomLength * boomChange + boomChange * boomChange;
     _boomLength = glm::clamp<float>(_boomLength, ZOOM_MIN, ZOOM_MAX);
+
+    // May need to change view if boom length has changed
+    if (fabs(boomChange) > 0.0f) {
+        qApp->changeViewAsNeeded(_boomLength);
+    }
```
Comment (AndrewMeadows) [inline interface/src/avatar/MyAvatar.cpp]: It looks `changeViewAsNeeded()` will be called even when `_boomLength` hasn't changed: e.g. when `boomChange` was non-zero but the `_boomLength` was clamped back to what it was.  Maybe we should do something like:
```
    float oldBoomLength = _boomLength;
    float boomChange = getDriveKey(ZOOM);
    _boomLength += 2.0f * _boomLength * boomChange + boomChange * boomChange;
    _boomLength = glm::clamp<float>(_boomLength, ZOOM_MIN, ZOOM_MAX);
    if (oldBoomLength != _boomLength) {
        qApp->changeViewAsNeeded(_boomLength);
    }
```

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10104/HighFidelity-Beta-PR13220-d01482bf687dd768adc612e58b195bcb0394ce15.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10104/HighFidelity-Beta-PR13220-d01482bf687dd768adc612e58b195bcb0394ce15.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10104/HighFidelity-Beta-Interface-PR13220-d01482bf687dd768adc612e58b195bcb0394ce15.dmg

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Camera modes work and reflect in the View menu.

**Pull request merged**